src/script/sigcache.h

```diff
@@ -70,7 +71,24 @@ class CachingTransactionSignatureChecker : public TransactionSignatureChecker
     CachingTransactionSignatureChecker(const CTransaction* txToIn, unsigned int nInIn, const CAmount& amountIn, bool storeIn, SignatureCache& signature_cache, PrecomputedTransactionData& txdataIn) : TransactionSignatureChecker(txToIn, nInIn, amountIn, txdataIn, MissingDataBehavior::ASSERT_FAIL), store(storeIn), m_signature_cache(signature_cache)  {}
 
     bool VerifyECDSASignature(const std::vector<unsigned char>& vchSig, const CPubKey& vchPubKey, const uint256& sighash) const override;
+
+    bool VerifySchnorrSignature(std::span<const unsigned char> sig, const XOnlyPubKey& pubkey, const uint256& sighash) const override;
+
+    bool GetStore() const { return store; }
+    SignatureCache& GetSigCache() const { return m_signature_cache; }
+};
+
+class CollectingSignatureChecker : public CachingTransactionSignatureChecker {
```

Comment:
https://github.com/bitcoin/bitcoin/pull/29491/commits/696bd7547e0e4ab44388d3c55ed636c46f9bdad5: IIUC the original idea was to "collect the signatures" and possibly combine them into one batch. Now that a "one batch per thread" approach has been taken, should this still be collecting signatures into a vector? the original "BatchSignatureChecker" that had a reference to the batch might work better since it skips the vector completely

Reply:
> the original "BatchSignatureChecker" that had a reference to the batch might work better since it skips the vector completely


On second thought, it might be better to leave as is because we are still experimenting and trying to find the right number  of batches to use